test/speech/index.js

```diff
@@ -0,0 +1,836 @@
+    });
+
+    describe('error', function() {
+      var error = new Error('Error.');
+      var RES = {};
+
+      beforeEach(function() {
+        speech.request = function(protoOpts, reqOpts, callback) {
+          callback(error, RES);
+        };
+      });
+
+      it('should execute callback with error & API response', function(done) {
+        speech.createRecognizeJob(REQ, function(err, operation, apiResponse) {
+          assert.strictEqual(err, error);
+          assert.strictEqual(operation, null);
+          assert.strictEqual(apiResponse, RES);
+          done();
+        });
+      });
+    });
+
+    describe('success', function() {
+      var RES = {
+        name: '1234'
+      };
+
+      beforeEach(function() {
+        speech.request = function(protoOpts, reqOpts, callback) {
+          callback(null, RES);
+        };
+      });
+
+      it('should execute callback with results & API resp', function(done) {
+        speech.createRecognizeJob(REQ, function(err, operation, apiResponse) {
+          assert.ifError(err);
+
+          assert.equal(operation.name, RES.name);
+          assert.equal(operation.results, undefined);
+          assert.strictEqual(apiResponse, RES);
+
+          done();
+        });
+      });
+    });
+  });
+
+  describe('createDetectJob', function() {
+    var findFiles_;
+    var REQ = {};
+
+    var FILES = [
+      {
+        content: 'aGk='
+      }
+    ];
+
+    before(function() {
+      findFiles_ = Speech.findFiles_;
+    });
+
+    beforeEach(function() {
+      Speech.findFiles_ = function(files, callback) {
+        callback(null, FILES);
+      };
+    });
+
+    after(function() {
+      Speech.findFiles_ = findFiles_;
+    });
+
+    it('should find the files', function(done) {
+      Speech.findFiles_ = function(files) {
+        assert.strictEqual(files, FILE);
+        done();
+      };
+
+      speech.createDetectJob(FILE, REQ, assert.ifError);
+    });
+
+    it('should return an error from findFiles_', function(done) {
+      var error = new Error('Error.');
+
+      Speech.findFiles_ = function(files, callback) {
+        assert.strictEqual(files, FILE);
+        callback(error);
+      };
+
+      speech.createDetectJob(FILE, REQ, function(err) {
+        assert.strictEqual(err, error);
+        done();
+      });
+    });
+
+    it('should return operation', function(done) {
+      speech.createRecognizeJob = function(REQ, callback) {
+        callback(null, speech.operation({
+          name: '1234'
+        }), {
+          name: '1234'
+        });
+      };
+
+      speech.createDetectJob(
+        FILE,
+        REQ,
+        function(err, response, apiResponse) {
+          assert.ifError(err);
+          assert.equal(response.name, '1234');
+          assert.deepEqual(apiResponse, {
+            name: '1234'
+          });
+          done();
+        }
+      );
+    });
+
+    it('should return an error from createRecognizeJob()', function(done) {
```

Comment:
I think we can safely remove this test, the only error we need to test for is one that comes from `Speech.findFiles_`.